### PR TITLE
[DownloadBuildArtifactsV1] Tar extraction on download

### PR DESCRIPTION
**Pull request by DaniilShmelev**
This PR repeats this changes made to V0: https://github.com/microsoft/azure-pipelines-tasks/pull/14579

This adds an `extractTars` boolean input. If this option is enabled, all downloaded .tar files will be extracted to $(targetDir)/extracted_tars, just like they would in V0 version of this task.

This feature is disabled in windows.

Initially, all archives are extracted to `$(Agent.TempDirectory)/extracted_tars`. This is done using `tar` system utility.
After that, the extracted_tars folder is moved to the target directory. All of this is handled by the `FileContainerProvider`.
Whether to extract the archives and the path to the extracted files is passed through `ArtifactDownloadParameters`.

### CODE REVIEW

File: src/Agent.Plugins/Artifact/FileContainerProvider.cs [hunk 1]
```diff
@@ -190,6 +206,34 @@ private async Task DownloadFileContainerAsync(ArtifactDownloadParameters downloa
             }
         }
 
+        private async Task<IEnumerable<FileContainerItem>> GetArtifactItems(ArtifactDownloadParameters downloadParameters, BuildArtifact buildArtifact)
```
Comment (anatolybolshakov) [inline src/Agent.Plugins/Artifact/FileContainerProvider.cs:209]: Could you please add some quick comments for new methods?

**Comment by anatolybolshakov** [comment]
Tested changes, LGTM!

**Review by EzzhevNikita — APPROVED** [review]
LGTM!
Tested this changes manually, found no issues

File: src/Agent.Plugins/Artifact/FileContainerProvider.cs (hunk 1 above)
Comment (DaniilShmelev) [reply src/Agent.Plugins/Artifact/FileContainerProvider.cs:210]: Added comments

File: src/Agent.Plugins/BuildArtifact/BuildArtifactPluginV1.cs [hunk 2]
```diff
@@ -73,6 +74,7 @@ public class DownloadBuildArtifactTaskV1_0_0 : BuildArtifactTaskPluginBaseV1
         static readonly string buildVersionToDownloadLatest = "latest";
         static readonly string buildVersionToDownloadSpecific = "specific";
         static readonly string buildVersionToDownloadLatestFromBranch = "latestFromBranch";
+        static readonly string extractedTarsTempDir = "extracted_tars";
```
Comment (anatolybolshakov) [inline src/Agent.Plugins/BuildArtifact/BuildArtifactPluginV1.cs:77]: Why do we need this temp dir?

Comment (DaniilShmelev) [reply src/Agent.Plugins/BuildArtifact/BuildArtifactPluginV1.cs:77]: This is how it worked in the V0 version of this task.
All downloaded .tar files are extracted to $(Agent.TempDir)/extracted_tars, and then copied over to target directory of this task.

**Review by alex-peck — APPROVED** [review]
LGTM

**Pull request merged**